src/pip/_internal/utils/entrypoints.py

```diff
@@ -25,3 +41,34 @@ def _wrapper(args: Optional[List[str]] = None) -> int:
         "running pip directly.\n"
     )
     return main(args)
+
+
+def get_best_invocation_for_this_pip() -> str:
+    """Try to figure out the best way to invoke pip in the current environment."""
+    binary_directory = "Scripts" if WINDOWS else "bin"
```

Comment:
I can't remember for certain - do we not support environments on Windows that use "bin" rather than "Scripts" (for example, MSYS2)? I note that `pip._internal.locations.get_bin_prefix()` allows both on Windows...

Reply:
I don't think we explicitly support them. None the less, if this fails, the thing that'd happen is a less-perfect prompt will be presented. :)

I've copied this over inspired by what `venv` does on 3.7 -- https://github.com/python/cpython/blob/3.7/Lib/venv/__init__.py#L114